hack/test-e2e.sh

```diff
@@ -14,138 +14,30 @@
-#               quickstart guide or an Envoy service IP based on the e2e test framework.
-#
-# WHAT THE SCRIPT DOES:
-#   1. Determines if there is a Gateway named "inference-gateway" in the "default" namespace. If found, it extracts the IP
-#      address and port from the Gateway's "llm-gw" listener. Otherwise, it falls back to the Envoy service in the "default" namespace.
-#   2. Optionally checks for (or creates) a "curl" pod, ensuring it is ready to execute requests.
-#   3. Loops for $TIME seconds, sending requests every 5 seconds to the /v1/completions endpoint to confirm successful connectivity.
-
-set -euo pipefail
-
-# Determine the directory of this script and build an absolute path to client.yaml.
-SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
-CLIENT_YAML="$SCRIPT_DIR/../test/testdata/client.yaml"
-
-# TIME is the amount of time, in seconds, to run the test.
-TIME=${TIME:-1}
-# Optionally use a client curl pod for executing the curl command.
-CURL_POD=${CURL_POD:-false}
-
-check_resource_exists() {
-    local type=$1
-    local name=$2
-    local namespace=$3
-
-    if kubectl get "$type" "$name" -n "$namespace" &>/dev/null; then
-         return 0
-    else
-         return 1
-    fi
-}
-
-check_pod_ready() {
-    local pod_name=$1
-    local namespace=$2
-    # Check the Ready condition using jsonpath. Default to False if not found.
-    local ready_status
-    ready_status=$(kubectl get pod "$pod_name" -n "$namespace" -o jsonpath='{.status.conditions[?(@.type=="Ready")].status}' 2>/dev/null || echo "False")
-    if [[ "$ready_status" == "True" ]]; then
-        return 0
-    else
-        return 1
-    fi
+set -euox pipefail
+
+install_kind() {
+  if ! command -v kind &>/dev/null; then
+    echo "kind not found, installing..."
+    [ $(uname -m) = x86_64 ] && curl -Lo ./kind https://kind.sigs.k8s.io/dl/v0.29.0/kind-linux-amd64
+    # For ARM64
+    [ $(uname -m) = aarch64 ] && curl -Lo ./kind https://kind.sigs.k8s.io/dl/v0.29.0/kind-linux-arm64
+    chmod +x ./kind
+    mv ./kind /usr/local/bin/kind
+  else
+    echo "kind is already installed."
+  fi
 }
 
-# Try to get the Gateway's IP and the port from the listener named "llm-gw" if it exists.
-if check_resource_exists "gateway" "inference-gateway" "default"; then
-    GATEWAY_IP=$(kubectl get gateway inference-gateway -n default -o jsonpath='{.status.addresses[0].value}')
-    # Use JSONPath to select the port from the listener with name "http"
-    GATEWAY_PORT=$(kubectl get gateway inference-gateway -n default -o jsonpath='{.spec.listeners[?(@.name=="http")].port}')
+if kubectl config current-context >/dev/null 2>&1; then
+  echo "Active kubecontext found. Running Go e2e tests in ./epp..."
 else
-    GATEWAY_IP=""
-    GATEWAY_PORT=""
+  install_kind
+  kind create cluster --name inference-e2e
+  KIND_CLUSTER=inference-e2e make image-kind
+  # Add the Gateway API CRDs
```

Comment:
I don't see the Gateway API CRDs being installed anywhere and they're not needed to run the e2e.